src/ResultSet.php

```diff
@@ -0,0 +1,74 @@
+<?php
+namespace CCR\Ralph;
+
+use IteratorAggregate, JsonSerializable;
+use function json_decode;
+
+/**
+ * A result set for working with results returned from a SPARQL endpoint.
+ */
+
+class ResultSet implements IteratorAggregate, JsonSerializable
+{
+    /** @var array Results returned from a query. */
+    private $results;
+
+    public function __construct(string $results)
+    {
+        $this->results = json_decode($results, true);
+    } // __construct()
+
+    /**
+     * Get the variables included in the results.
+     *
+     * @return array The variables as an array.
+     */
+
+    public function getVariables(): array
+    {
+        return $this->results['head']['vars'];
```

Comment:
What if these elements aren't set? Should you return `false` or `null` or throw an exception in that case?